Wiggle: build failure with `tracing: false` with async functions

### Test Case

Add `tracing: false` to the `from_witx!()` definition in `crates/wiggle/tests/atoms_async.rs`

### Steps to Reproduce

`cargo test -p wiggle`

### Expected Results

Successful test run

### Actual Results

```
error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> crates/wiggle/tests/atoms_async.rs:8:1
   |
8  | / wiggle::from_witx!({
9  | |     witx: ["$CARGO_MANIFEST_DIR/tests/atoms.witx"],
10 | |     async: *,
11 | |     tracing: false,
12 | | });
   | |  ^
   | |  |
   | |__only allowed inside `async` functions and blocks
   |    this is not `async`
   |
   = note: this error originates in the macro `wiggle::from_witx` (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Versions and Environment

Wasmtime version or commit: Current HEAD, 387426e7f.

Operating system: macOS 12.6

Architecture: x86_64

--- a/crates/wiggle/generate/src/funcs.rs
+++ b/crates/wiggle/generate/src/funcs.rs
@@ -118,7 +118,11 @@ fn _define_func(
                 }.instrument(_span)
             )
         } else {
-            quote!(#body)
+            quote!(
+                async move {
+                    #body
+                }
+            )
         };
         (
             quote!(

--- a/crates/wiggle/tests/tracing.rs
+++ b/crates/wiggle/tests/tracing.rs
@@ -1,0 +1,16 @@
+// This just tests that things compile when `tracing: false` is set,
+// which isn't the default.
+
+wiggle::from_witx!({
+    witx: ["$CARGO_MANIFEST_DIR/tests/atoms.witx"],
+    async: {
+        atoms::double_int_return_float,
+    },
+    tracing: false,
+});
+
+impl wiggle::GuestErrorType for types::Errno {
+    fn success() -> Self {
+        types::Errno::Ok
+    }
+}

